pkg/test/utils.go

```diff
@@ -165,7 +165,19 @@ func CompareGoldenObject(t *testing.T, path string, got []byte) {
 	if err != nil {
 		t.Errorf("Failed parsing file: %s", err)
 	}
-	if diff := cmp.Diff(wantMap, gotMap); diff != "" {
-		t.Errorf("unexpected diff in %s: %s", path, diff)
+
+	diff := cmp.Diff(wantMap, gotMap)
+	if diff == "" {
+		return
+	}
+
+	if writeGoldenOutput {
+		// Write the output to the golden file
+		if err := os.WriteFile(p, []byte(got), 0644); err != nil {
+			t.Fatalf("failed to write golden output %s: %v", p, err)
+		}
+		t.Errorf("wrote updated golden output to %s", p)
```

Comment:
Since `writeGoldenOutput ` is true, it's expected we are writing/updating the golden output. So use `Logf` here is clearer than `Errorf`. Otherwise, I think we want to make the message more specific about why we consider this an error. For example: "diff detected between incoming log and golden log. Wrote updated golden log to %s"